add Emlalock support and fix keylength problem

BondageBob opened the pull request.

embeddedkink committed:
Update index.html

embeddedkink committed:
update uploader

embeddedkink committed:
minor warning changes

embeddedkink commented:
Thank you for the merge request! I've only had a short look at the code. I understand that it doesn't check whether the session was ended successfully or if it was aborted, is that correct? I would (eventually) like to see the box keep track of the estimated end time, and fall back on that when a session is aborted. 
I'll get on merging when I have time, but that may take a few weeks. In the future, please send different issues as separate requests, that would make it easier for example to fast-track the key length issue.

BondageBob commented:
sorry for mixing both feature in one PR, but it's two commits you could cherry-pick the key fix.

I could track the `failedsessions` field and keep the box locked until `enddate`, but I would assume that `enddate` doesn't hold information if the user isn't allowed to see his remaining time.

embeddedkink committed:
Hotfix for blocking build error

embeddedkink committed:
possible performance improvement, api more flexible

embeddedkink committed:
update installer

BondageBob committed:
add emlalock api support

the vault will lock when the user is in an active session.

BondageBob committed:
prevent the use of keys with more than 64 chars

It was possible to lock the box with a key longer than 64 characters,
but you could only use 64 characters for the unlock key. This kept the
box locked forever.

MeisterBob committed:
replacedeprecated  DynamicJsonDocument with JsonDocument

Signed-off-by: BondageBob <git@gerhard-joerges.de>

MeisterBob committed:
logging and some minor tweaks

MeisterBob committed:
run clang-format on all files
